python/lsst/ts/attcpip/at_simulator.py

```diff
@@ -0,0 +1,246 @@
+
+        json_schema = registry[payload_id]
+        try:
+            jsonschema.validate(data, json_schema)
+        except jsonschema.ValidationError as e:
+            self.log.exception("Validation failed.", e)
+            return False
+        return True
+
+    async def _write_command_response(self, response: str, sequence_id: int) -> None:
+        """Generic method to write a command response.
+
+        Parameters
+        ----------
+        response : `str`
+            The response to write. Acceptable responses are defined in the
+            ``Ack`` enum. The value of response is not checked.
+        sequence_id : `int`
+            The command sequence id.
+        """
+        data = {
+            CommonCommandArgument.ID: response,
+            CommonCommandArgument.SEQUENCE_ID: sequence_id,
+        }
+        await self.cmd_evt_server.write_json(data=data)
+
+    async def write_ack_response(self, sequence_id: int) -> None:
+        """Write an ``ACK`` response.
+
+        Parameters
+        ----------
+        sequence_id : `int`
+            The command sequence id.
+        """
+        await self._write_command_response(Ack.ACK, sequence_id)
+
+    async def write_fail_response(self, sequence_id: int) -> None:
+        """Write a ``FAIL`` response.
+
+        Parameters
+        ----------
+        sequence_id : `int`
+            The command sequence id.
+        """
+        await self._write_command_response(Ack.FAIL, sequence_id)
+
+    async def write_noack_response(self, sequence_id: int) -> None:
+        """Write a ``NOACK`` response.
+
+        Parameters
+        ----------
+        sequence_id : `int`
+            The command sequence id.
+        """
+        await self._write_command_response(Ack.NOACK, sequence_id)
+
+    async def write_success_response(self, sequence_id: int) -> None:
+        """Write a ``SUCCESS`` response.
+
+        Parameters
+        ----------
+        sequence_id : `int`
+            The command sequence id.
+        """
+        await self._write_command_response(Ack.SUCCESS, sequence_id)
+
+    async def _write_evt(self, evt_id: str, **kwargs: typing.Any) -> None:
+        """Write an event message.
+
+        Parameters
+        ----------
+        evt_id : `str`
+            The name of the event, for instance ``evt_atMountState``.
+        kwargs : `typing.Any`
+            The data to include in the event message.
+        """
+        data: dict[str, typing.Any] = {CommonCommandArgument.ID: evt_id, **kwargs}
+        await self.cmd_evt_server.write_json(data=data)
+
+    async def _write_telemetry(
+        self, tel_id: str, data: typing.Any, timestamp: np.float64 | None = None
+    ) -> None:
+        """Write a telemetry message.
+
+        Parameters
+        ----------
+        tel_id : `str`
+            The name of the telemetry.
+        data : `typing.Any`
+            The data to include in the telemetry message.
+        timestamp : `np.float64` | None
```

Comment:
`np.float64` or `float`?

Reply:
`np.float64` is the data type that is sent to this function by the implementing simulators.